recipe-server/normandy/settings.py

```diff
@@ -110,6 +111,25 @@ class Core(Configuration):
         }
     }
 
+    # Content Security Policy
+    def CSP_DEFAULT_SRC(self):
+        srcs = ["'self'"]
+        if self.CDN_URL:
+            srcs.append(self.CDN_URL)
+        return srcs
+    CSP_OBJECT_SRC = "'none'"  # not using <object>, <embed>, and <applet> elements
+    CSP_WORKER_SRC = "'none'"  # not using JS Worker, SharedWorker, or ServiceWorkers
+    CSP_FRAME_SRC = "'none'"  # not using frames or iframes
+
+    # these don't fallback to default-src
+    CSP_BASE_URI = "'none'"  # not using <base>
+    CSP_FRAME_ANCESTORS = "'none'"  # this page isn't iframed elsewhere
+    CSP_FORM_ACTION = "'self'"  # we only submit forms to ourselves
+
+    # when https is enabled
+    CSP_UPGRADE_INSECURE_REQUESTS = True
+    CSP_BLOCK_ALL_MIXED_CONTENT = True
```

Comment:
Actually, reading the documentation a little closer, we probably just want `CSP_BLOCK_ALL_MIXED_CONTENT`. It's not recommended to have both of these enabled, as the former treats http pages as https for the purposes of CSP, and the latter blocks http resources on https pages. Having both enabled makes the mixed content check do nothing.

I think having mixed content on for all of our environments works fine; we shouldn't have environments that are both http and https, they should be one or the other.